doc/source/rllib/metrics-logger.rst

```diff
@@ -0,0 +1,431 @@
+    RLlib controls all of these "hand over" points entirely, so unless you write your own subcomponent that reports to a parent component, such as
+    :py:class:`~ray.rllib.algorithms.algorithm.Algorithm`, you should refrain from calling the :py:meth:`~ray.rllib.utils.metrics_logger.MetricsLogger.reduce`
+    method on any :py:class:`~ray.rllib.utils.metrics_logger.MetricsLogger`.
+
+    To get the current reduced result, use the :py:meth:`~ray.rllib.utils.metrics_logger.MetricsLogger.peek` method instead,
+    which doesn't alter any underlying values.
+
+
+Instead of providing a flat key, you can also log a value under some nested key through passing in a tuple:
+
+.. testcode::
+
+    # Log a value under a deeper nested key.
+    logger.log_value(("some", "nested", "key"), -1.0)
+    print(logger.peek(("some", "nested", "key")))  # expect: -1.0
+
+
+To use reduce methods, other than "mean", specify the ``reduce`` argument in
+:py:meth:`~ray.rllib.utils.metrics_logger.MetricsLogger.log_value`:
+
+.. testcode::
+
+    # Log a maximum value.
+    logger.log_value(key="max_value", value=0.0, reduce="max")
+
+Because you didn't specify a ``window`` and are using ``reduce="max"``, the window used is the infinite window,
+meaning MetricsLogger reports the lifetime maximum value, whenever reduction takes place or you peek at the current value:
+
+.. testcode::
+
+    for i in range(1000, 0, -1):
+        logger.log_value(key="max_value", value=float(i))
+
+    logger.peek("max_value")  # expect: 1000.0, which is the lifetime max (infinite window)
+
+
+You can also chose to not reduce at all, but to simply collect individual values, for example a set of images you receive
+from your environment over time and for which it doesn't make sense to reduce them in any way.
+
+Use the ``reduce=None`` argument for achieving this. However, it's strongly advised that you should also
+set the ``clear_on_reduce=True`` flag, because this may cause memory leaks otherwise.
+This flag assures that :py:class:`~ray.rllib.utils.metrics_logger.MetricsLogger` clears out the underlying list of values after every
+``reduce()`` handover operation, for example from :py:class:`~ray.rllib.env.env_runner.EnvRunner`
+to :py:class:`~ray.rllib.algorithms.algorithm.Algorithm`:
+
+.. testcode::
+
+    logger.log_value("some_items", value="a", reduce=None, clear_on_reduce=True)
+    logger.log_value("some_items", value="b")
+    logger.log_value("some_items", value="c")
+    logger.log_value("some_items", value="d")
+
+    logger.peek("some_items")  # expect a list: ["a", "b", "c", "d"]
+
+    logger.reduce()
+    logger.peek("some_items")  # expect an empty list: []
+
+
+Logging a set of nested scalar values
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In case you are logging a nested structure of values, for example
+``{"time_s": 0.1, "lives": 5, "rounds_played": {"player1": 10, "player2": 4}}`` and all values have the exact same log settings
```

Comment:
A dict that is for any reason empty, will be reduced away, which is for collecting result dicts for a table is challenging.

Reply:
Ok, let me try this right now and fix, if possible ...

Reply:
Yeah, an empty dict is not logged at all. Meaning ...
```
metrics.log_dict({}, key="abc")
```
... results in `metrics` not undergoing any changes, which I think makes a lot of sense. There are simply no keys to be logged and the underlying `tree.map_structure` doesn't have any nested structure to loop through.